### PR TITLE
TYP: misc typing from pandas-stubs in io and plotting

### PR DESCRIPTION
**Pull request by twoertwein**

### CODE REVIEW

File: pandas/io/parsers/readers.py [hunk 1]
```diff
@@ -434,15 +436,19 @@
...
 
 
-_c_parser_defaults = {
-    "delim_whitespace": False,
-    "na_filter": True,
-    "low_memory": True,
-    "memory_map": False,
-    "float_precision": None,
-}
+class _c_parser_defaults(Enum):
```
Comment (twoertwein) [inline pandas/io/parsers/readers.py:439]: For a TypedDict we would have needed a declaration and then the actual dict - using an Enum is much simpler here.

Comment (twoertwein) [reply pandas/io/parsers/readers.py]: Switch to the more verbose TypedDict: duplicate values in the enum

File: pandas/io/formats/style.py [hunk 2]
```diff
@@ -3390,8 +3393,11 @@ def highlight_quantile(
 
     @classmethod
     def from_custom_template(
-        cls, searchpath, html_table: str | None = None, html_style: str | None = None
-    ):
+        cls,
+        searchpath: Sequence[str],
```
Comment (twoertwein) [inline pandas/io/formats/style.py:3397]: also accepts strings based on the stubs

File: pandas/io/parsers/readers.py [hunk 3]
```diff
@@ -599,25 +615,25 @@ def read_csv(
...
+    converters: Mapping[Hashable, Callable] | None = ...,
+    true_values: list | None = ...,
+    false_values: list | None = ...,
     skipinitialspace: bool = ...,
-    skiprows=...,
+    skiprows: list[int] | int | Callable[[Hashable], bool] | None = ...,
     skipfooter: int = ...,
     nrows: int | None = ...,
-    na_values=...,
+    na_values: Sequence[str] | Mapping[str, Sequence[str]] | None = ...,
```
Comment (twoertwein) [inline pandas/io/parsers/readers.py:628]: Seems to also allow a str

**Review by mroeschke — COMMENTED** [review]
Looks okay, but there's a merge conflict

**Comment by twoertwein** [comment]
rebased + green

**Comment by twoertwein** [comment]
green again

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @twoertwein

**Pull request merged**